Sources/JWTProvider/PayloadAuthenticationMiddleware.swift

```diff
@@ -64,22 +64,24 @@ public final class PayloadAuthenticationMiddleware<U: PayloadAuthenticatable>: M
 
         let filteredSigners = try self.signers(for: jwt)
 
-        // Try to use all the signers until one matches
-        var verified = false
+        // Default Error
+        var error: Swift.Error? = JWTError.signatureVerificationFailed
 
+        // Try to use all the signers until one matches
         for signer in filteredSigners {
 
             do {
-                _ = try req.jwt(verifyUsing: signer, and: claims)
-                verified = true
+                _ = try req.jwt(verifyUsing: signer, and: self.claims)
+                error = nil
                 break
-            } catch {
+            } catch let jwtError {
+                error = jwtError
                 continue
             }
         }
 
-        guard verified else {
-            throw JWTProviderError.noVerifiedJWT
```

Comment:
The problem with the previous implementation is that `noVerifiedJWT` overrides the jwt validation error.

It could fail because the token is invalid (which should end up in a 401) , or because claims are not validated (403).

Also, can't throw an array of errors, you'll need an error, that can contains many errors. In that case, which one of them would define the status code?